src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -93,6 +90,24 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
     Record<string, React.ReactNode>
   >({});
 
+  const [defaultDataSourceId, setDefaultDataSourceId] = useState<string | null>();
```

Comment:
Nit: we can separate a hook something like `useScopedDefaultDataSource` to share the loading logic with `edit_data_source.tsx` in the future.